compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -677,8 +677,6 @@ static LogicalResult setDefaultRootConfig(
     func::FuncOp entryPointFn,
     PartitionableLoopsInterface partitionableLoopsInterfaceOp,
     ArrayRef<int64_t> lbs, ArrayRef<int64_t> ubs) {
-  if (getLoweringConfig(partitionableLoopsInterfaceOp)) return success();
-
```

Comment:
Could we preserve this for safety or are we planning to override an existing config?

Reply:
Or maybe an assert?

Reply:
Maybe I can walk through all the compute ops before setting config? In this case, we won't need to add the assertion everywhere. What do you think?

Reply:
If the pre-condition of each setter function is that there shouldn't be a configuration already, I think it's better to assert in each function to avoid accidental bugs if the code change in the future.

Reply:
This is what I was referring to in my comment (not just user defined configuration). Doing a walk for every config setting is expensive..... Assert is OK, but I would rather error out....

Reply:
The walk only happens before setting the configuration for a root op, which is cheap. The code structure makes your concerns hard to happen to me. Because we can only set a lowering_config on the root op.

I'm actually fine with not checking it in every function. They are all static methods, not functions in utils. We can make some assumptions since they are only used in this file. I think adding assertions is enough. It is very wrong to me if it happens.